numpy/ma/core.py

```diff
@@ -191,14 +191,19 @@ class MaskError(MAError):
     default_filler["M8[" + v + "]"] = np.datetime64("NaT", v)
     default_filler["m8[" + v + "]"] = np.timedelta64("NaT", v)
 
+float_types_list = [np.half, np.single, np.double, np.longdouble,
+                    np.csingle, np.cdouble, np.clongdouble]
+complex_filler = np.array([np.inf+0j, -np.inf-0j])
+complex_filler.imag = np.array([np.inf, -np.inf])
 max_filler = ntypes._minvals
-max_filler.update([(k, -np.inf) for k in [np.float32, np.float64]])
+max_filler.update([(k, -np.inf) for k in float_types_list[:4]])
+max_filler.update([(k, complex_filler[0]) for k in float_types_list[-3:]])
+
 min_filler = ntypes._maxvals
-min_filler.update([(k, +np.inf) for k in [np.float32, np.float64]])
-if 'float128' in ntypes.typeDict:
-    max_filler.update([(np.float128, -np.inf)])
-    min_filler.update([(np.float128, +np.inf)])
+min_filler.update([(k,  +np.inf) for k in float_types_list[:4]])
+max_filler.update([(k, complex_filler[1]) for k in float_types_list[-3:]])
```

Comment:
```suggestion
max_filler.update([(k, complex(+np.inf, +np.inf)) for k in float_types_list[-3:]])
```
I don't know, but seems slightly nicer to me. Is fine as it was though! It does change the type to the python type... But in the end that is probably actually just as well or better.